class-mb-rest-api.php

```diff
@@ -200,6 +230,44 @@ public function update_comment_meta( $data, $object ) {
 		do_action( 'rwmb_after_save_post', $object->comment_ID );
 	}
 
+	/**
+	 * Get settings page meta for the rest API.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success or WP_Error object on failure.
+	 */
+	public function get_settings_meta( WP_REST_Request $request ) {
+		$settings_pages_id = $request->get_param( 'id' );
+		if ( ! $settings_pages_id ) {
+			return [];
+		}
+
+		$meta_boxes = rwmb_get_registry( 'meta_box' )->get_by( [ 'object_type' => 'setting' ] );
+		$meta_boxes = array_filter( $meta_boxes, function( $meta_box ) use ( $settings_pages_id ) {
+			return in_array( $settings_pages_id, $meta_box->settings_pages, true );
+		} );
+
+		$option_name = $this->get_option_name_from_settings_page_id( $settings_pages_id );
+		return $this->get_values( $meta_boxes, $option_name, ['object_type' => 'setting'] );
+	}
+
+	private function get_option_name_from_settings_page_id( $settings_pages_id ) {
+		$settings_pages = apply_filters( 'mb_settings_pages', [] );
+		$settings_pages = array_filter( $settings_pages, function ( $settings_page ) use ( $settings_pages_id ) {
+			return $settings_pages_id === $settings_page['id'];
+		} );
```

Comment:
Đừng có lạm dụng `array_filter`. Ý nghĩa của `filter` là lọc, trong khi ở đây mình cần là tìm. Dùng `for...each` cho đơn giản.